CMake/UseITK.cmake

```diff
@@ -228,8 +239,11 @@ foreach(_factory_name ${ITK_FACTORY_LIST})
 endforeach()
 #-----------------------------------------------------------------------------
 if(NOT ITK_NO_IO_FACTORY_REGISTER_MANAGER)
-
   set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ITK_IO_FACTORY_REGISTER_MANAGER)
   include_directories(BEFORE ${CMAKE_CURRENT_BINARY_DIR}/ITKFactoryRegistration)
+endif()
 
+if(NOT ITK_NO_FFT_FACTORY_REGISTER_MANAGER)
+  set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ITK_FFT_FACTORY_REGISTER_MANAGER)
+  include_directories(BEFORE ${CMAKE_CURRENT_BINARY_DIR}/ITKFactoryRegistration)
```

Comment:
For new code, I suggest we avoid associating CMake properties like `COMPILE_DEFINITIONS` and `INCLUDE_DIRECTORIES` to directories.

Instead, these should be associated with relevant ITK CMake targets.That way, linking against a given target is the only requirement to ensure the relevant compilation flags et al are propagated either internally or when linking against ITK.

Reply:
For some background information, see https://docs.google.com/document/d/11b4grg4mtovEiRroiijomMinKba6J25nbJaDpyiqw-4/edit#

Reply:
Thanks for weighing in @jcfr. This sounds like the best path for expansion in the future, but it doesn't look like `itk_configure_factory` has been implemented yet and I'm worried about expanding the scope of this PR too far. Rather than introducing more CMake changes than we have to here, maybe it would be best to rely on existing architecture for now and use FFT classes as a case study to assist in implement the new macro in a subsequent change?

As a side note, the document listed is dated from 2016, was there a reason that `itk_configure_factory` was not implemented or just lack of project bandwidth?

Reply:
> just lack of project bandwidth?

Exactly.

>  I'm worried about expanding the scope of this PR too far.

That is sensible.

Reply:
Marking as resolved for now, can explore further discussion on Discourse.